src/core/lib/security/credentials/tls/grpc_tls_certificate_verifier.cc

```diff
@@ -210,9 +210,11 @@ int grpc_tls_certificate_verifier_verify(
 
 void grpc_tls_certificate_verifier_cancel(
     grpc_tls_certificate_verifier* verifier,
-    grpc_tls_custom_verification_check_request* request) {
+    grpc_tls_custom_verification_check_request* request, grpc_status_code code,
+    const char* error_details) {
   grpc_core::ExecCtx exec_ctx;
-  verifier->Cancel(request);
+  verifier->Cancel(request, absl::Status(static_cast<absl::StatusCode>(code),
+                                         error_details));
```

Comment:
nit: To be safe, we should probably check that `error_details` is not nullptr before passing into the `absl::Status`, e.g. we could write `error_details ? error_details : ""`.